Update documentation for GitLab MR comments (#213)
returntocorp/semgrep-docs: docs/notifications.md

## GitLab merge request comments

Merge request comments are currently supported for gitlab.com users only. [Sign up for the GitLab beta](https://go.r2c.dev/join-gitlab-beta) if you have a self-hosted instance of GitLab and are interested in receiving MR comments.

Automated comments on GitLab merge requests look like this:

<p style="text-align: center; font-size: 12px">
    <img width="600px" src="../img/gitlab-mr-comment.png" alt="Screenshot of a GitLab MR comment"/><br/>
    An inline GitLab merge request comment left by a custom Semgrep rule
</p>

To enable MR comments:

1. Log into Semgrep's [Dashboard > Settings](https://semgrep.dev/manage/settings) to obtain your deployment ID and an API token.

> 2. Examine your [policy settings](https://semgrep.dev/manage/policies/starter-policy?tab=settings) and make sure that PR comments are enabled. PR comments are the same thing as MR comments.

Reviewer: ```suggestion
2. Examine your [policy settings](https://semgrep.dev/manage/policies/starter-policy?tab=settings) and make sure that PR comments are enabled (MR comments and PR comments are enabled by the same toggle switch).
```

Note: this link didn’t load for me (I got a spinner but it never updated the main body of the page): https://semgrep.dev/manage/policies/starter-policy?tab=settings

Author: @pabloest yes, this is a tough judgment call. New users get a policy called "starter-policy" but older users will have "getting-started" or something similar. We could instead here just link to semgrep.dev/manage/policies, which will work for everyone, but it won't bring them right to the policy settings page - they have to then click an individual policy and navigate to the "settings" tab.

Reviewer: @chmccreery ah makes sense. I’d leave the starter-policy then.

Author: In other words, I am happy to either stick with the link above, which will work for new users and not work for old users, or use /manage/policies, which will work for everyone but require two more clicks that may not be super obvious, so require more instructions here probably.

Reviewer: I’m in favor of optimizing for new users by keeping the link you’re using in this PR.

Author: Oops, somehow didn't see your comments before now! Feel free to change back!! Sorry @pabloest

Author: Oops, somehow didn't see your comments before now! Feel free to change back!! Sorry @pabloest

Author: https://github.com/returntocorp/semgrep-docs/pull/214

> 2. Examine your [policy settings](https://semgrep.dev/manage/policies/starter-policy?tab=settings) and make sure that PR comments are enabled (MR comments and PR comments are enabled by the same toggle switch).

Author: ```suggestion
2. Examine your policy settings by navigating to [Dashboard > Policies](https://semgrep.dev/manage/policies), selecting a policy, and switching to the Settings tab. Make sure that PR comments are enabled (MR comments and PR comments are enabled by the same toggle switch).
```

3. Create an API token on gitlab.com by going to [Profile > Access Tokens](https://gitlab.com/-/profile/personal_access_tokens) and adding a token with `api` scope.
4. Copy the token that GitLab gives you.

> 5. Navigate to your repository's CI/CD settings, scroll down to `Variables`, and click 'expand'. The url will look like https://gitlab.com/myusername/myproject/-/settings/ci_cd.

Reviewer: ```suggestion
5. Navigate to your repository's Settings > CI/CD, scroll down to 'Variables', and click 'Expand'. The url will look like https://gitlab.com/myusername/myproject/-/settings/ci_cd.
```

> 6. Give the new variable the key `PAT` and use the token you copied in step 2 as the value. Select "mask variable" and **UNSELECT "protect variable"**.

Reviewer: ```suggestion
6. Click to 'Add variable', give the new variable the key `PAT` and use the token you copied in step 2 as the value. Select "mask variable" and **UNSELECT "protect variable"**.
```
